### PR TITLE
HOCS-4288 added support for getting users with their associated team …

### PR DESCRIPTION
**Pull request by LaurieBennett**
…uuids

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/info/api/TeamResource.java [hunk 1]
```diff
@@ -148,6 +152,25 @@ public ResponseEntity deleteTeamPermissions(@PathVariable String teamUUID, @Requ
...
+    public ResponseEntity<Set<UserWithTeamsDto>> getUsersForTeam() {
+        List<UserDto> users = userService.getAllUsers();
+        Set<UserWithTeamsDto> usersWithTeams = new HashSet<>();
+
+        for (UserDto user: users){
+            Set<Team> teams = teamService.getTeamsForUser(UUID.fromString(user.getId()));
+            ArrayList<UUID> teamUUIDS = new ArrayList<>();
+            teams.forEach((team -> {
+                teamUUIDS.add(team.getUuid());
+                team.getUnit().getTeams().forEach((unitTeam) -> {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/api/TeamResource.java:165]: Not sure I understand this - is this the case that a user is in team1 for unit1 and thus has all teams from unit1?

Comment (LaurieBennett) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/TeamResource.java:165]: This seemed to make sense to me, it's possible this is wrong, originally I tried to pass the teamDTO, but due to infinite recursion this broke the response, so I'm simply adding the teams and their units as this seemed to be more useful than just returning the team's uuid and asking the user to then retrieve the correct units on their own, I can change this to just returning the team's uuid quickly if that seems like a better idea

Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/TeamResource.java]: If this shows what it should I'm happy to accept for now. Still reads slightly odd but I'll add it for something to look at.

**Review by LiamMacP — APPROVED** [review]
Looks fine to me, would be good to re-evaluate the team and unit link. As this is really for internal and users won't be informed of this endpoint at this point, I'm happy to check that aftet this is merged.

**Pull request merged**